packages/libisitweeka/src/index.ts

```diff
@@ -143,43 +145,222 @@ export default class IsItWeekA {
 
 		const ics = await baseResponse.text();
 
-		// console.log(ics);
+		// New Version!
+
+		console.log(ics);
 
 		const data = ical.parseICS(ics);
 
-		// console.log(data);
+		console.log(data);
+
+		const mapWithoutRecurrences = new Map(Object.entries(data));
+
+		const map = new Map();
+
+		// Process recurrences and add to the Map.
+		mapWithoutRecurrences.forEach((event, key) => {// When dealing with calendar recurrences, you need a range of dates to query against,
+			// Add every event in the Map into the one with recurrences.
+			map.set(key, event);
+
+			// because otherwise you can get an infinite number of calendar events.
+			var rangeStart = new Date("2020-01-01");
+			var rangeEnd = new Date("2030-12-31");
+
+			if (event.type === "VEVENT") {
+				let title = event.summary;
+				if (typeof event.start === "undefined") {
+					console.error("event.start undefined:", title);
+				}
+				let startDate = event.start || new Date();
+				if (typeof event.end === "undefined") {
+					console.error("event.end undefined:", title, event.start?.toISOString());
+				}
+				let endDate = event.end || new Date();
+
+				// Calculate the duration of the event for use with recurring events.
+				let duration = endDate.getTime() - startDate.getTime();
+
+				// Simple case - no recurrences, just print out the calendar event.
+				if (typeof event.rrule === 'undefined') {
+					if (title?.toLowerCase().startsWith("week") && startDate.getFullYear() === 2022 && startDate.getMonth() > 6) {
+						console.log("title:", title);
+						console.log("startDate:", startDate);
+						console.log("endDate:", endDate);
+						console.log("duration:", duration);
+						console.log("event type:", "non-recurrence");
+						console.log();
+					}
+				}
+
+				// Complicated case - if an RRULE exists, handle multiple recurrences of the event.
+				else if (typeof event.rrule !== 'undefined') {
+					// For recurring events, get the set of event start dates that fall within the range
+					// of dates we're looking for.
+					console.warn(event);
+					console.warn(event.summary);
+					console.warn(event.rrule);
+					console.warn(typeof event.rrule);
+					console.warn(event.rrule.between);
+					const partialFixedRRule = rrule.fromString(event.rrule as unknown as string);
```

Comment:
Should we add a if (typeof event.rrule !== "string") { return; } ?

Reply:
Because what if a "Week A" non-recursive event is added? Looks like the code would error?

Reply:
it wouldn't, because non-recursive events don't have a `rrule` property and are handled by the above `typeof event.rrule === "undefined"`

The only possible type of `event.rrule` at this point is a string (even though the d.ts file for `ical` suggests otherwise)

Reply:
Ah ok 👍